libs/platform/core/src/services/storage/default-indexed-db-storage.service.ts

```diff
@@ -1,52 +1,73 @@
-import { Dexie, liveQuery } from 'dexie';
-import { Observable, shareReplay, switchMap } from 'rxjs';
+import { subscribeReplay } from '@spryker-oryx/utilities';
+import { map, Observable, shareReplay, Subscriber, switchMap } from 'rxjs';
 import {
   indexedDbStorageName,
   IndexedDBStorageService,
   indexedDbTableName,
 } from './indexed-db-storage.service';
-import { StoredValue } from './model';
 
+/** @deprecated since 1.2 */
 export class DefaultIndexedDBStorageService implements IndexedDBStorageService {
-  protected storage = new Observable<Dexie>((subscriber) => {
-    const db = new Dexie(indexedDbStorageName);
-    db.version(1).stores({
-      [indexedDbTableName]: '&key,value',
-    });
-
-    subscriber.next(db);
-
-    return () => {
-      db.close();
-    };
-  }).pipe(
-    switchMap(async (db) => {
-      await db.open();
-
-      return db.table(indexedDbTableName);
-    }),
-    shareReplay({ refCount: false, bufferSize: 1 })
-  );
-
-  getItem(key: string): Observable<StoredValue> {
-    return this.storage.pipe(
-      switchMap((storage) =>
-        liveQuery<StoredValue>(async () => {
-          return (await storage.get(key))?.value;
-        })
+  protected db$: Observable<IDBDatabase> = new Observable(
+    (subscriber: Subscriber<IDBDatabase>) => {
+      const request = indexedDB.open(indexedDbStorageName, 1);
+
+      request.onerror = () => subscriber.error('DB: error');
+      request.onupgradeneeded = (evt: IDBVersionChangeEvent) => {
```

Comment:
I think we don't need to save 2 letters :) Was not able to find in our code base such an abbreviation.
```suggestion
      request.onupgradeneeded = (event: IDBVersionChangeEvent) => {
```